specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2019-05-01/NasCluster.json

```diff
@@ -0,0 +1,207 @@
+        "operationId": "NasClusters_Get",
+        "parameters": [
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/NasClusterParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/NasCluster"
+            }
+          },
+          "404": {
+            "description": "NOT FOUND"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Fabric.Admin/fabricLocations/{location}/nasClusters": {
+      "get": {
+        "x-ms-examples": {
+          "Returns a list of all storage nas clusters at a location.": {
+            "$ref": "./examples/NasCluster/List.json"
+          }
+        },
+        "tags": [
+          "NasClusters"
+        ],
+        "description": "Returns a list of all nas clusters at a location.",
+        "operationId": "NasClusters_List",
+        "parameters": [
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../2016-05-01/Fabric.json#/parameters/FilterParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/NasClusterList"
+            }
+          },
+          "404": {
+            "description": "NOT FOUND"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/NasCluster"
+      }
+    }
+  },
+  "definitions": {
+    "NasCluster": {
+      "description": "Representation of a nas cluster resource.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "Properties of a nas cluster.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NasClusterModel"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../2016-05-01/Fabric.json#/definitions/Resource"
+        }
+      ]
+    },
+    "NasClusterModel": {
+      "description": "Properties of a nas cluster.",
+      "type": "object",
+      "properties": {
+        "clusterName": {
```

Comment:
>clusterName [](start = 9, length = 11)

Is this different than the resource name?

Reply:
Yes. As the cluster name is input by user. We need to ensure its unique so generate GUID for that.

Reply:
Sounds good.